schematic/schemas/data_model_parser.py

```diff
@@ -267,24 +267,29 @@ def parse_entry(
         rel_entry: Any,
         id_jsonld_key: str,
         dn_label_dict: dict[str, str],  # pylint:disable=unused-argument
-        model_jsonld: dict,
+        model_jsonld: list[dict],
     ) -> Any:
         """Parse an input entry based on certain attributes
 
         Args:
-            rel_entry (Any): Given a single entry and relationship in a JSONLD data model,
-              the recorded value
-            id_jsonld_key (str): the jsonld key for id
-            dn_label_dict (_type_): Not used
-            model_jsonld (dict): _description_
-
+            rel_entry: Given a single entry and relationship in a JSONLD data model, the recorded value
+            id_jsonld_key, str: the jsonld key for id
+            dn_label_dict, dict[str:str]:dict of model labels to display names
+            model_jsonld, list[dict]: list of dictionaries, each dictionary is an entry in the jsonld data model
         Returns:
             Any: n entry that has been parsed base on its input type and
               characteristics.
         """
-        # Retrieve ID from single value dictionary
-        if isinstance(rel_entry, dict) and len(rel_entry.keys()) == 1:
-            parsed_rel_entry = rel_entry["@id"]
+        # Parse dictionary entries
+        if isinstance(rel_entry, dict):
+            # Retrieve ID from single value dictionary
+            if set(rel_entry.keys()) == {"@id"}:
+                parsed_rel_entry = rel_entry["@id"]
+            # Parse any remaining dictionaries
+            else:
+                parsed_rel_entry = self.convert_entry_to_dn_label(
+                    rel_entry, model_jsonld
```

Comment:
THis block is only if `rel_entry` is a dict, but `convert_entry_to_dn_label` only handles strings or lists.

Reply:
Okay, I thought through the logic of what was happening, and 1. moved into its own helper, and am now just returning the dictionary.